devlib/utils/android.py

```diff
@@ -445,6 +445,24 @@ def adb_command(device, command, timeout=None,adb_server=None):
     output, _ = check_output(full_command, timeout, shell=True)
     return output
 
+def grant_app_permissions(target, package):
+    """
+    Grant an app all the permissions it may ask for
+    """
+    dumpsys = target.execute('dumpsys package {}'.format(package))
+
+    permissions = re.search(
+        'requested permissions:\s*(?P<permissions>(android.permission.+\s*)+)', dumpsys
+    )
+    permissions = permissions.group('permissions').replace(" ", "").splitlines()
+
+    for permission in permissions:
+        try:
+            target.execute('pm grant {} {}'.format(package, permission))
+        except:
```

Comment:
It would be good if this didn't catch everything blindly. What is the error you expect to see here?

Reply:
A quick run with exoplayer raises:

```
TargetError: Got exit code 1
from target command: pm grant com.google.android.exoplayer2.demo android.permission.INTERNET
STDOUT: 
STDERR: Operation not allowed: java.lang.SecurityException: Permission android.permission.INTERNET is not a changeable permission type
```

I think it's safe enough to search for 'java.lang.SecurityException' in stderr and propagate the exception if it's not found.

Reply:
That might break when future version of Android decide to rename the exception or format the error differently. I think catching `TargetError` here should be sufficient, as this wraps a single operation, and programming errors in the underlying implementation will manifest as some other exception type.